metsim/disaggregate.py

```diff
@@ -7,107 +7,179 @@
 import itertools
 import scipy
 
-from metsim.configuration import PARAMS as params
-from metsim.configuration import CONSTS as consts
+import metsim.constants as cnst
+from metsim.physics import svp
 
-def disaggregate(df_daily, solar_geom):
+def disaggregate(df_daily, params, solar_geom):
     """
     TODO
     """
     stop = params['stop'] + pd.Timedelta('1 days')     
     dates_disagg = pd.date_range(params['start'], stop, freq=params['time_step']+'T')
     df_disagg = pd.DataFrame(index=dates_disagg)
+    n_days = len(df_daily['day_of_year'])
+    n_disagg = len(df_disagg.index)
+    ts = float(params['time_step'])
+    
     df_disagg['shortwave'] = (shortwave(df_daily['swrad'],
-                                       df_daily['dayl'],
-                                       df_daily['day_of_year'],
-                                       solar_geom['tiny_rad_fract']))
-    df_disagg['temp'] = temp(df_daily, df_disagg)
-    df_disagg['precip'] = precip(df_daily['precip'])
-    df_disagg['longwave'] = longwave(df_daily['lwrad'])
-    df_disagg['wind'] = wind(df_daily['wind'])
+                                        df_daily['dayl'],
+                                        df_daily['day_of_year'],
+                                        solar_geom['tiny_rad_fract'],
+                                        params))
+    
```

Comment:
Each call to a function in the `disaggregate` module will return the disaggregated values for the variable named by the function.   Documentation is a huge issue with this codebase right now, I'll admit.